src/modules/keyboardmanager/common/KeyboardManagerState.h

```diff
@@ -21,35 +22,46 @@ class KeyboardManagerState
 private:
     // State variable used to store which UI window is currently active that requires interaction with the hook
     KeyboardManagerUIState uiState;
+    std::mutex uiState_mutex;
 
     // Window handle for the current UI window which is active. Should be set to nullptr if UI is deactivated
     HWND currentUIWindow;
+    std::mutex currentUIWindow_mutex;
 
     // Vector to store the shortcut detected in the detect shortcut UI window. This is used in both the backend and the UI.
     std::vector<DWORD> detectedShortcut;
+    std::mutex detectedShortcut_mutex;
 
     // Store detected remap key in the remap UI window. This is used in both the backend and the UI.
     DWORD detectedRemapKey;
+    std::mutex detectedRemapKey_mutex;
 
     // Stores the UI element which is to be updated based on the remap key entered.
     TextBlock currentSingleKeyRemapTextBlock;
+    std::mutex currentSingleKeyRemapTextBlock_mutex;
 
     // Stores the UI element which is to be updated based on the shortcut entered
     TextBlock currentShortcutTextBlock;
+    std::mutex currentShortcutTextBlock_mutex;
 
 public:
+    // The map members and their mutexes are left as public since the maps are used extensively in dllmain.cpp
     // Maps which store the remappings for each of the features. The bool fields should be initalised to false. They are used to check the current state of the shortcut (i.e is that particular shortcut currently pressed down or not).
     // Stores single key remappings
     std::unordered_map<DWORD, WORD> singleKeyReMap;

```

Comment:
nit: Not related to the Current PR. Is there any specific reason for storing mapping as `DWORD`(32 bit) to `WORD`(16 bit) similarly in case of shortcut remappings map?

Reply:
Initially I had done it like that since the hook APIs would require the virtual key code arguments as `DWORD` and SendInput would use `WORD` so I did that to reduce warnings, but now with all the different cases I still had to add many type casts to suppress the warnings. I will refactor it to be DWORD everywhere since it seems to be making its making the code more confusing now.